Mage/src/main/java/mage/abilities/keyword/PrototypeAbility.java

```diff
@@ -1,17 +1,68 @@
 package mage.abilities.keyword;
 
+import mage.ObjectColor;
 import mage.abilities.SpellAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.condition.common.PrototypedCondition;
+import mage.abilities.costs.mana.ManaCost;
 import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.decorator.ConditionalContinuousEffect;
+import mage.abilities.effects.common.continuous.SetBasePowerToughnessSourceEffect;
+import mage.abilities.effects.common.continuous.SetCardColorSourceEffect;
+import mage.abilities.effects.common.cost.SetCardCostSourceEffect;
 import mage.cards.Card;
+import mage.constants.*;
+import mage.game.Game;
+import mage.util.CardUtil;
+
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.stream.Collectors;
 
 /**
- * @author TheElk801
+ * @author TheElk801, Susucr
  */
 public class PrototypeAbility extends SpellAbility {
 
-    public PrototypeAbility(Card card, String manaString, int power, int toughness) {
-        super(new ManaCostsImpl<>(manaString), card.getName());
-        // TODO: implement this
+    private int power;
+    private int toughness;
+
+    public PrototypeAbility(Card card, String manaString, ObjectColor color, int power, int toughness) {
```

Comment:
why do we need to pass in color here? the mana cost should be enough info for that